### DISCUSSION TITLE
targetHandle not working

### DISCUSSION THREAD

**Discussion by Basiiil** [xyflow/xyflow#4155] [General]
i use this sample https://codesandbox.io/p/sandbox/exciting-shaw-tyfq4v?file=%2Fsrc%2Fstyles.css from #2027
and change this
{
    id: "a1-a3",
    source: "A-1",
    target: "A-3",
    sourceHandle: "r1",
    targetHandle: "l1"
  }
to this:
{
    id: "a1-a3",
    source: "A-1",
    target: "A-3",
    sourceHandle: "r1",
    targetHandle: "b1"
  }

but not working!!
help meee please

**Comment by bcakmakoglu** [comment]
The problem you're facing is that your `<Handle>` component don't use the `type` prop defining whether they are source or target handles, which makes them fall back to the default which is `source`.
By default the connection mode that's used is *strict*, meaning that only a source -> target (and vice versa) edge is allowed.

You can omit the `type` prop by setting `connectionMode={ConnectionMode.Loose}` which allows source -> source or target -> target.

[Here's an example.](https://codesandbox.io/p/sandbox/wild-wildflower-jrk62t?file=%2Fsrc%2FApp.js%3A69%2C49)

**Reply by Basiiil** [reply]
yesss,thanks alot
